openapi3/internalize_refs.go

```diff
@@ -2,18 +2,22 @@ package openapi3
 
 import (
 	"context"
+	"path"
 	"path/filepath"
 	"strings"
 )
 
-type RefNameResolver func(string) string
+type RefNameResolver func(*T, componentRef) string
 
 // DefaultRefResolver is a default implementation of refNameResolver for the
 // InternalizeRefs function.
 //
 // If a reference points to an element inside a document, it returns the last
 // element in the reference using filepath.Base. Otherwise if the reference points
 // to a file, it returns the file name trimmed of all extensions.
+//
+// Deprecated: Use UpdatedDefaultRefNameResolver which is less likely to generate
```

Comment:
No, let's go with the API breakage. Just add an entry to the changelog section of the README

Reply:
Sure,

Reply:
Done

Reply:
https://github.com/getkin/kin-openapi/pull/955/commits/23f97c6de03b179de8df6cc55aaa51681d172758